firmware/hw_layer/sensors/gps_uart.cpp

```diff
@@ -30,42 +28,43 @@ static THD_WORKING_AREA(gpsThreadStack, UTILITY_THREAD_STACK_SIZE);
 
 // this field holds our current state
 static loc_t GPSdata;
+static efidatetime_t lastDateTime;
 
-static int gpsMesagesCount = 0;
+static int gpsMessageCount = 0;
 static int uartErrors = 0;
 
-// todo: some data structure for coordinates location
-// todo:
+// TODO: some data structure for coordinates location
+
 float getCurrentSpeed(void) {
+    // TODO: ???
 	return GPSdata.speed;
 }
 
 static void printGpsInfo() {
 	efiPrintf("GPS RX %s", hwPortname(engineConfiguration->gps_rx_pin));
 	efiPrintf("GPS TX %s", hwPortname(engineConfiguration->gps_tx_pin));
 
-	efiPrintf("m=%d,e=%d: vehicle speed = %.2f", gpsMesagesCount, uartErrors, getCurrentSpeed());
+	efiPrintf("m=%d,e=%d: vehicle speed = %.2f", gpsMessageCount, uartErrors, getCurrentSpeed());
 
 	float sec = getTimeNowMs() / 1000.0;
-	efiPrintf("communication speed: %.2f", gpsMesagesCount / sec);
+	efiPrintf("communication speed: %.2f", gpsMessageCount / sec);
 
 	efiPrintf("GPS latitude = %.2f\r\n", GPSdata.latitude);
 	efiPrintf("GPS longitude = %.2f\r\n", GPSdata.longitude);
 }
 
-static struct tm curTm;
-
-static void onGpsMessage(char *buffer) {
-
+static void onGpsMessage(const char * const buffer) {
 	gps_location(&GPSdata, buffer);
-	date_get_tm(&curTm);
 
-	if (GPSdata.quality == 4 && GPSdata.GPStm.tm_year > 0 && GPSdata.GPStm.tm_sec != curTm.tm_sec) {
-		// quality =4 (valis GxRMC), year > 0, and difference more then second
-		date_set_tm(&GPSdata.GPStm);					// set GPS time
```

Comment:
@rusefillc this comes from `rtc_helper`, whose API changed, hence the bundling with the other changes in #4764

Reply:
yes, that one is already in master but https://github.com/rusefi/rusefi/pull/4764/commits/a00ac0bf038809707a87dad2091595dd6d294981 was a bad example for making the point about atomic changes